### PR TITLE
Changed retrieve_tails to delete invalid tails files after fetching

### LINKED ISSUE #1196: Remove invalid/corrupted tails files
When acapy attempts to download a tails file (e.g. for revoking/issuing a credential), it will attempt to download the file from the public URI. If this process fails, (e.g. the tails server is temporarily unavailable and answers with an error code), then acapy will download the file, check for a valid hash (this check fails) and does not remove the tails file if an invalid hash value is detected.

The next time the tails file is required, acapy will use the invalid file in the filesystem (resulting in errors in indy-sdk/ursa) instead of attempting a redownload.

### PR DESCRIPTION
**Pull request by c2bo**
Remove downloaded tails files with invalid hashes. This fixes #1196

### CODE REVIEW

**Comment by andrewwhitehead** [comment]
Hiya, tests are currently failing because we have strict Python code formatting rules enabled and lines 179 and 180 are too long. You should be able to install and run the 'black' Pypi package to resolve the issue automatically.

File: aries_cloudagent/revocation/models/revocation_registry.py [hunk 1]
```diff
@@ -169,23 +170,33 @@ async def retrieve_tails(self):
...
-                raise RevocationError(
-                    "The hash of the downloaded tails file does not match."
-                )
+        download_tails_hash = base58.b58encode(file_hasher.digest()).decode("utf-8")
+        if download_tails_hash != self.tails_hash:
+            try:
+                os.remove(tails_file_path)
+                tails_file_dir.rmdir()
+            except OSError:
+                LOGGER.warning(f"Could not delete invalid tails file: {OSError}")
```
Comment (andrewwhitehead) [inline aries_cloudagent/revocation/models/revocation_registry.py:195]: OSError is a class instance, that's probably not going to print the desired result. You might want something like `except OSError as err:`

**Comment by andrewwhitehead** [comment]
Sorry about the delay, this looks good apart from my one comment. I can't add changes to the branch so you'll need to rebase the branch until we can get it merged.

**Comment by c2bo** [comment]
@andrewwhitehead Rebased to main and fixed the error handling for file removal

**Review by domwoe — APPROVED** [review]

**Review by ianco — APPROVED** [review]

**Pull request merged**